test/testcommon/src/main/java/io/pravega/test/common/NoOpScheduledExecutor.java

```diff
@@ -9,110 +9,173 @@
     }
 
     @Override
     public ScheduledFuture<?> scheduleWithFixedDelay(Runnable runnable, long l, long l1, TimeUnit timeUnit) {
-        return null;
+        return new DummyScheduledFuture(0);
     }
 
     @Override
     public void shutdown() {
-
+        isShutDown.set(true);
     }
 
     @Override
     public List<Runnable> shutdownNow() {
-        return null;
+        isShutDown.set(true);
+        return new ArrayList<>();
     }
 
     @Override
     public boolean isShutdown() {
-        return false;
+        return isShutDown.get();
     }
 
     @Override
     public boolean isTerminated() {
-        return false;
+        return isShutDown.get();
     }
 
     @Override
-    public boolean awaitTermination(long l, TimeUnit timeUnit) throws InterruptedException {
-        return false;
+    public boolean awaitTermination(long l, TimeUnit timeUnit) {
+        return isShutDown.get();
     }
 
     @Override
     public <T> Future<T> submit(Callable<T> callable) {
-        return null;
+        return CompletableFuture.completedFuture(null);
     }
 
     @Override
     public <T> Future<T> submit(Runnable runnable, T t) {
-        return null;
+        return CompletableFuture.completedFuture(null);
     }
 
     @Override
     public Future<?> submit(Runnable runnable) {
-        return null;
+        return CompletableFuture.completedFuture(null);
     }
 
     @Override
-    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> collection) throws InterruptedException {
-        return null;
+    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> collection) {
+        return new ArrayList<>();
     }
 
     @Override
-    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> collection, long l, TimeUnit timeUnit) throws InterruptedException {
-        return null;
+    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> collection, long l, TimeUnit timeUnit) {
+        return new ArrayList<>();
     }
 
     @Override
-    public <T> T invokeAny(Collection<? extends Callable<T>> collection) throws InterruptedException, ExecutionException {
-        return null;
+    public <T> T invokeAny(Collection<? extends Callable<T>> collection) {
+        throw new UnsupportedOperationException();
     }
 
     @Override
-    public <T> T invokeAny(Collection<? extends Callable<T>> collection, long l, TimeUnit timeUnit) throws InterruptedException, ExecutionException, TimeoutException {
-        return null;
+    public <T> T invokeAny(Collection<? extends Callable<T>> collection, long l, TimeUnit timeUnit) {
+        throw new UnsupportedOperationException();
     }
 
     @Override
     public void execute(Runnable runnable) {
+    }
+}
+
+@EqualsAndHashCode // Required to keep Spotbugs satisfied (owing to code in `compareTo()`).
+@RequiredArgsConstructor
+class DummyScheduledFuture implements ScheduledFuture<Integer> {
```

Comment:
Hmm.. I wonder how come checkstyle didn't catch this. We shouldn't define more than one class/interface per file. 

You should move this as a static nested class inside the `NoOpScheduledExecutor` class.

Reply:
Yeah, that sounds good. Done.